docs/basic-component.md

```diff
@@ -0,0 +1,247 @@
+# Basic component recipe
+
+Recipe for creating a basic component.
+
+- [Component structure](#component-structure)
+  - [Subcomponents](#subcomponents)
+  - [Hooks and Utils](#hooks-and-utils)
+  - [index.ts](#indexts-1)
+  - [Component.tsx](#componenttsx)
+  - [Styles](#styles)
+  - [Tests](#tests)
+- [Component.tsx template](#componenttsx-template)
+  - [Props](#props)
+  - [The component](#the-component)
+  - [Display name](#display-name)
+
+## Component structure
+
+Given the component `PancakeStack`, the component structure will follow this:
```

Comment:
Where would we put common types used between components?

Reply:
That is a very good question... Perhaps we'd need to have a separate package just for those generic types. I was certainly thinking we could introduce a type which deals with the `className` rename eg.
```
interface ClassNameOverride<TBaseAttributes> extends Omit<TBaseAttributes, "className"> { 
  classNameOverride?: string
}
```
assuming it works.

Which now that I look at it, we need to figure out if we are happy to prefix generics with a `T` or if that's just a me thing haha.